R/resample_to_field.R

```diff
@@ -42,7 +44,7 @@ resample_to_field <- function(nc_file, field_file, method = 'match', precision =
     mutate(Depth = as.numeric(Depth))
   
   # join model results to observations
-  validation = field_obs %>% left_join(model.wide, by = c("DateTime", "Depth"))
+  validation = field_obs %>% inner_join(model.wide, by = c("DateTime", "Depth"))
```

Comment:
Old method was like inner_join. I was mistaken initially because I forgot there was a step truncating dates. 

However an option to left or inner joint could be useful if you wanted to match it up with your original observation data frame. Ie return the same number of rows.

Reply:
Old method was like inner_join. I was mistaken initially because I forgot there was a step truncating dates. 

However an option to left or inner joint could be useful if you wanted to match it up with your original observation data frame. Ie return the same number of rows.

Reply:
Old method was like inner_join. I was mistaken initially because I forgot there was a step truncating dates. 

However an option to left or inner joint could be useful if you wanted to match it up with your original observation data frame. Ie return the same number of rows.

Reply:
Was it like inner_join? I thought you got NAs back when the obs existed at deeper depths than the model had, which is not the behavior you'd get from an inner join.

Reply:
So there are two differences in the new method (current implementation) and the old method
1) What you wrote above. Old method returns NA for depths deeper than the model (ex. Obs go down to 25 m, but model only to 20 m). 
2) Old method will return NA for dates that are within model start/stop dates but observations fall on dates don't have model output. This would only happen if the model output wasn't daily. 

The new method is a true inner join. So not going to have any NAs. If I was coding this model from scratch, I would probably have a join argument with a `left_join` or `inner_join` option. That way you could return a data.frame the same length as your observed data frame with a bunch of NAs, or return an NA-free data.frame.

Reply:
Thanks - ok, the NA depth part matches my understanding on it too. 

For the NAs when dates don't overlap - I thought that is what the `precision` and `match` args were for. Does that create an issue with that part of the implementation?

Reply:
Probably good to add `tidyr >= 1.0.0` to the DESCRIPTION file since I think `pivot_longer` doesn't exist in earlier versions